packages/fluentui/react-northstar/src/components/Carousel/Carousel.tsx

```diff
@@ -243,52 +252,44 @@ class Carousel extends AutoControlledComponent<WithAsProp<CarouselProps>, Carous
       activeIndex = 0;
     }
 
-    this.setState({
-      prevActiveIndex: this.state.activeIndex,
-      activeIndex,
-    });
+    actions.UpdatePreActiveIndex(activeIndex);
+    actions.UpdateActiveIndex(activeIndex);
```

Comment:
There is naming collision in this scope

```suggestion
    actions.UpdateActiveIndex(nextActiveIndex);
```

Reply:
I also suggest to have a single action for these properties

Reply:
Done